### DISCUSSION TITLE
Can i add client-readonly traits?

### DISCUSSION THREAD

**Discussion by sergeyshaykhullin** [ory/kratos#1004] [Q&A] [answered]
Can I add a read-only trait on the client? And be able to change it only through the admin api?

**Comment by k9ert** [comment, marked as answer]
As the [documentation](https://www.ory.sh/kratos/docs/concepts/identity-data-model/#identity-traits-and-json-schemas) points outm 

> They are also supposed to be modified by the identity itself e.g. as part of the registration or profile update process as well as
> anyone having access to ORY Krato's Admin API.

There are plans to add "system" traits but i'm not sure there is any chance to get a reasonable release-date for that feature.
Sorry, but you will need to deal with that information elsewhere.

**Reply by aeneasr** [reply]
To expand @k9ert's answer, progress is tracked in https://github.com/ory/kratos/issues/47 . However, we are currently working on the other things, particularly MFA and so on.
